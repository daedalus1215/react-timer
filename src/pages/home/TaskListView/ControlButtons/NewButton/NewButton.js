import React from 'react';
import cn from 'classnames';
import { fetchApiData } from 'utils';
import { useBrowserHistory } from 'hooks';
import { Button } from 'components';
import hydrateTaskForm from '../../../TaskForm/hooks/useSubmit/hydrateTaskForm';
import styles from './NewButton.module.css';

/**
 * We will create a new task before we start working on the note.
 *
 * @param {Function} push function to the next url
 */
const createNewTask = push => () => {
  const initialTask = {
    id: -1,
    description: '',
    tags: [],
    project: 0,
    time: 0,
  };

  const emptyTask = hydrateTaskForm({ id: -1 }, [], initialTask);
  const dispatch = data => {
    push(`/task/${data._id}`);
    window.location.reload();
  }

  fetchApiData('task', { body: emptyTask, method: 'POST' }, dispatch);
};

const NewButton = () => {
  const { push } = useBrowserHistory();
  const addNewTask = createNewTask(push);

  return <Button data-test-id="btn-new"
    className={cn(styles.buttonAdd, "glyphicon glyphicon-edit")}
    title="New Task"
    onClick={addNewTask} />
};

export default NewButton;