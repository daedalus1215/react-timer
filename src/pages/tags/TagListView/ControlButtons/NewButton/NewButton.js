import React from 'react';
import cn from 'classnames';
import { Button } from 'components';
import { useDispatch } from 'react-redux';
import styles from './NewButton.module.css';
import { postTag } from 'redux/actionCreators/actions';

const NewButton = () => {
  const dispatch = useDispatch();

  return <Button data-test-id="btn-new"
    className={cn(styles.buttonAdd, "glyphicon glyphicon-edit")}
    onClick={() => {
      dispatch(postTag({ description: "", name: "" }));
      window.location.reload();
    }} />;
};

export default NewButton;