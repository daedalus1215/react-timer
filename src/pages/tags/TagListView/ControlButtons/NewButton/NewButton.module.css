.buttonAdd,
.buttonAdd:hover {
  background-color: rgb(150, 150, 150);
  color: white;
  border: none;
  text-decoration: none;
  text-align: center;
  width: 45px;
  height: 40px;
  margin: 4px 5px;
  border-radius: 4px;
}

.buttonAdd:hover {
  background-color: #a4413a;
}
