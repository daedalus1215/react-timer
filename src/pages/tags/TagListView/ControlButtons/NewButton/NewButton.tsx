import React from 'react';
import cn from 'classnames';
import { Button } from 'components';
import { useDispatch } from 'react-redux';
import styles from './NewButton.module.css';
import { postTag } from 'redux/actionCreators/actions';

const NewButton: React.FC = () => {
  const dispatch = useDispatch();

  return <Button data-testid="btn-new"
    className={cn(styles.buttonAdd, "glyphicon glyphicon-edit")}
    onClick={() => {
      console.log('yes')
      dispatch(postTag({ description: "", name: "" }));
    }} />;
};

export default NewButton;