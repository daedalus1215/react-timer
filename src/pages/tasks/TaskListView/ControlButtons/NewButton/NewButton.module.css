.buttonAdd,
.buttonAdd:hover {
  background-color: rgb(150, 150, 150);
  color: white;
  width: 45px;
  height: 40px;
  margin: 4px 5px;
}

.buttonAdd:hover {
  background-color: rgb(50, 50, 50);
}
