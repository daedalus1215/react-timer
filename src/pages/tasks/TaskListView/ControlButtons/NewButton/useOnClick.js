import { useBrowserHistory } from 'hooks';
import { fetchApiData, getCurrentDateTimeEstFormat } from 'utils';
import hydrateTaskForm from '../../../TaskForm/hooks/useSubmit/hydrateTaskForm';

const useOnClick = () => {
    const { push } = useBrowserHistory();

    return () => {
        const emptyTask = hydrateTaskForm(-1, [], 0, '', getCurrentDateTimeEstFormat(), 0, []);
        const dispatch = data => {
            push(data?._id ? `/task/${data._id}` : '/');
            window.location.reload();
        }

        fetchApiData('task', { body: emptyTask, method: 'POST' }, dispatch);
    };
};

export default useOnClick;