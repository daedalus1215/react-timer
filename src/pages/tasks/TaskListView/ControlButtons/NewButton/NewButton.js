import React from 'react';
import cn from 'classnames';
import { Button } from 'components';
import useOnClick from './useOnClick';
import styles from './NewButton.module.css';

const NewButton = () => {
  const addNewTask = useOnClick();

  return <Button
    testid="btn-new"
    className={cn(styles.buttonAdd, "glyphicon glyphicon-edit")}
    title="New Task"
    onClick={addNewTask} />
};

export default NewButton;