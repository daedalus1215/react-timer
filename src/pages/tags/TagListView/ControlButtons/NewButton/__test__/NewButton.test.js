import React from 'react';
import NewButton from '../NewButton';
import { render } from '@testing-library/react';

jest.mock('hooks/useBrowserHistory');
jest.mock('redux/actionCreators/actions');
jest.mock('react-redux');

describe('src/pages/tags/TagsPage/TagListViewControlPanel/NewButton/__test__/NewButton.test.js', () => {
  describe('NewButton', () => {
    describe('NewButton', () => {
      it('should display NewButton', async () => {
        // Arrange & Act
        const { findByTestId } = render(<NewButton />);

        // Assert
        expect(await findByTestId('btn-new')).toMatchSnapshot();
      });
    });
  });
});