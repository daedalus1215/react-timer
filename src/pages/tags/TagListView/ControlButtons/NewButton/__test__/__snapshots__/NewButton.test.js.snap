// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/tags/TagsPage/TagListViewControlPanel/NewButton/__test__/NewButton.test.js NewButton NewButton should display NewButton 1`] = `
<button
  class="btn btn-default navbar-btn baseBtn buttonAdd glyphicon glyphicon-edit"
  data-testid="btn-new"
/>
`;
